src/matrix_solver.py

```diff
@@ -12,8 +12,12 @@ def solve_matrix(A, b):
     """
 
     import numpy as np
-    from numpy.linalg import solve
+    import numpy.linalg as linalg
 
-    temperature_array = solve(A, b)
+    try:
+        i = np.linalg.inv(A)
```

Comment:
This line takes substantially more work than solving the system Ax = b and `i` is not used.

Reply:
Is this meant as a test for singularity of the matrix?  [Here is a Stackoverflow discussion](https://stackoverflow.com/questions/13249108/efficient-pythonic-check-for-singular-matrix) on how to do this.